src/physics.cpp

```diff
@@ -1127,6 +1127,11 @@ void inelastic_scatter(const Nuclide& nuc, const Reaction& rx, Particle& p)
 
   // evaluate yield
   double yield = (*rx.products_[0].yield_)(E_in);
+  // Kill incident particle if yield is zero
+  if (yield == 0) {
+    p.wgt() = 0.;
+    return;
+  }
   if (std::floor(yield) == yield) {
```

Comment:
Might simplify things to just rely on the `else` branch to set the weight to zero:
```suggestion
  if (std::floor(yield) == yield && yield > 0) {
```

Reply:
It is indeed equivalent and much simpler. I have applied the change.